library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java

```diff
@@ -146,46 +156,100 @@ private void applyStyle() {
         binding.etMessage.setTypeface(Typeface.DEFAULT, style.getInputTextStyle());
     }
 
-    private void init() {
-        binding.setActiveMessageSend(false);
-        binding.ivSend.setOnClickListener(this);
-        binding.ivOpenAttach.setOnClickListener(this);
-        binding.etMessage.setOnFocusChangeListener(this);
-        binding.etMessage.addTextChangedListener(this);
-        binding.etMessage.setCallback((InputContentInfoCompat inputContentInfo,
-                                       int flags, Bundle opts) -> {
-            if (BuildCompat.isAtLeastQ()
-                    && (flags & InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
-                try {
-                    inputContentInfo.requestPermission();
-                } catch (Exception e) {
-                    return false;
+    private void configOnClickListener(){
+        binding.ivSend.setOnClickListener(view ->
+                onSendMessage(binding.etMessage.getText().toString(), viewModel.isEditing())
+        );
+        binding.ivOpenAttach.setOnClickListener(view -> {
+            binding.setIsAttachFile(true);
+            if (style.isPermissionSet() || isGrantedPermissions())
+                messageInputController.onClickOpenBackGroundView(MessageInputType.ADD_FILE);
+            else if(permissionRequestListener != null)
+                permissionRequestListener.openPermissionRequest();
+        });
+    }
+
+    private void configInputEditText(){
+        binding.etMessage.setOnFocusChangeListener((View view, boolean hasFocus)-> {
+            viewModel.setInputType(hasFocus ? InputType.SELECT : InputType.DEFAULT);
+            if (hasFocus) {
+                lockScrollUp = true;
+                new Handler().postDelayed(() -> lockScrollUp = false, 500);
+                Utils.showSoftKeyboard((Activity) getContext());
+            } else
+                Utils.hideSoftKeyboard((Activity) getContext());
+        });
+
+        binding.etMessage.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                //noop
```

Comment:
If you want to use just one method of an interface (`TextWatcher` in this case)  there is a way to make a little bit nicer for reading and save some space:
```java
// imported static method and lambda
afterTextChanged(binding.etMessage, editable -> {
      int l = editable.length();
      l++;
});
```
With functional interfaces and static utility class:
```java
interface Actions {
    interface EditableAction{
        void act(Editable editable);
    }
}

static class TestUtils {
    static void afterTextChanged(TextView textView, Actions.EditableAction action){
        textView.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                //ignore
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                //ignore
            }

            @Override
            public void afterTextChanged(Editable editable) {
                action.act(editable);
            }
        });
    }
}
```